keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -382,10 +390,100 @@ class ClusterHandler(
     return spec
   }
 
-  private fun List<ResourceDiff<ServerGroup>>.createsNewServerGroups() =
-    any {
-      !it.isCapacityOrAutoScalingOnly()
+  override suspend fun actuationInProgress(resource: Resource<ClusterSpec>) =
+    resource
+      .spec
+      .locations
+      .regions
+      .map { it.name }
+      .any { region ->
+        orcaService
+          .getCorrelatedExecutions("${resource.id}:$region")
+          .isNotEmpty()
+      }
+
+  /**
+   * Retrieve server group entity tags from clouddriver to identity the "source" of the server group (pipeline or
+   * orchestration), then retrieve execution details from orca to determine the deployment strategy used for that
+   * server group.
+   */
```

Comment:
Do you think this function could be moved to a generic class of its own, and then used for both ec2 and titus? That would help keep this long bit of logic out of the ec2 specific cluster handler, and would make it easier to split up into smaller functions if that was helpful.

Reply:
Yes, totally! I completely forgot about Titus. 🤦‍♂️